### ISSUE TITLE
(vscode) Format deletes comments in struct

### ISSUE THREAD

**Issue by Naddiseo** [rust-analyzer/rust-analyzer#6634]
```rust
struct Foo {
    hello: i8,
    /*
    line
    line
    */
    // line
    field1: u8,
    field2: u8,
    //hello
}
```
Formats as:
```rust

struct Foo {
    hello: i8,
    field1: u8,
    field2: u8,
    //hello
}
```

Not 100% sure this is the correct repository I should be reporting in. But I have removed the "rust" extension, and I have the rust-analyzer set as nightly (rust-analyzer version: nightly (9b512f8)), and also use the extension as the default formatter.

Please let me know if there's other information/diagnostics you need.

**Comment by bjorn3** [comment]
Does this also happen on `cargo fmt`? If so, please report it at https://github.com/rust-lang/rustfmt/issues.

**Comment by Naddiseo** [comment]
Issue resolved in rustfmt repo: my default system toolchain had be pinned to an old version and I'd misunderstood, or forgotten, how it worked. Setting the default toolchain to "stable" has fixed my problem.
